### PR TITLE
feat(config): Allow for version ranges

### LINKED ISSUE #3: Node version range
I don't want to lock down the version of node on my build. Could I instead do a version range?

Happy to make a PR if you're game :+1:

Something like:

``` json
"release": {
  "verifyConditions": {
    "path": "condition-node-version",
    "node": "^4",
    "verbose": true // optional
  }
}
```

So in my `.travis.yml` I can do:

```
node_js:
  - '4'
  - '0.12'
  - '0.10'
```

Thoughts?

### PR DESCRIPTION
**Pull request by kentcdodds**
Closes #3

### CODE REVIEW

**Comment by bahmutov** [comment]
I will fix the unit test before releasing.
Also, this might lead to tricky errors if several node versions match. For example if you test with 4.2.4 and 4.4.0 and release when matches '4' - both versions will cause a release which will screw up the npm publish

**Comment by kentcdodds** [comment]
Yeah, I think it's up to the user to make sure the version range they specify only matches one node version.

**Comment by kentcdodds** [comment]
Odd the tests failed. Seemed to work fine on my machine :-/

File: index.js [hunk 1]
```diff
@@ -14,7 +16,7 @@ function conditionNodeVersion (pluginConfig, config, callback) {
     return fail('Missing node version in the config')
   }
 
-  if (process.versions.node !== pluginConfig.node) {
+  if (semver.satisfies(process.versions.node, pluginConfig.node)) {
```
Comment (bahmutov) [inline index.js:19]: wait, I think this should be `if (!semver.satisfies(...`

Comment (kentcdodds) [inline index.js:19]: 😅 whoops...

**Comment by bahmutov** [comment]
Ok, published under https://github.com/bahmutov/condition-node-version/releases/tag/v1.3.0
Thanks for your contribution @kentcdodds

**Comment by kentcdodds** [comment]
Thank you for the package!

**Comment by kentcdodds** [comment]
Out of curiosity, what's your release process? I would have expected my pull request merge to have kicked off a release. How did you avoid that?

**Comment by bahmutov** [comment]
it uses semantic release of course, but your merge had unit test failure, so only I have fixed it it went through automatically.

**Comment by kentcdodds** [comment]
Oh, of course. Cool.

**Pull request merged**